bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{Error, Event, LOG_TARGET};
+use codec::DecodeAll;
+use core::marker::PhantomData;
+use snowbridge_core::Channel;
+use snowbridge_router_primitives::inbound::envelope::Envelope;
+use snowbridge_router_primitives::inbound::{MessageProcessor, VersionedXcmMessage};
+use sp_runtime::DispatchError;
+
+pub struct XcmMessageProcessor<T>(PhantomData<T>);
```

Comment:
I would probably call it some other thing but it is likely that snowbridge will tell you how to name it. So for me this is good enough

Reply:
Reason why I named it `XcmMessageProcessor` as it expects incoming message to be converted to xcm.  :) But, let's see what snowbridge thinks.